File: valid-parenthesis.js
const isValid = (string) => {
  let s = [];
  let output = true;
  let index = 0;
  let opens = "([{";
  let closes = ")]}";

  while (index < string.length && output) {
    let sym = string[index];
    if (opens.includes(sym)) {
      s.push(sym);
    } else {
      if (s.length === 0) {
        output = false;
      } else {
        let top = s.pop();
        if (opens.indexOf(top) != closes.indexOf(sym)) {
          output = false;
        }
      }
    }
    index++;
  }

  if (output && s.length === 0) {
    return true;
  } else {
    return false;
  }
};

console.log(isValid("({[})"));

File: valid-parenthesis.py
def isValid(string):
    s = []
    output = True
    index = 0
    opens = "([{"
    closes= ")]}"

    while index < len(string) and output:
        sym = string[index]
        if sym in opens:
            s.append(sym)
        else:
            if len(s) == 0:
                output = False
            else:
                top = s.pop()
                if not (opens.index(top) == closes.index(sym)):
                    output = False
        index +=1 
    if output and len(s) == 0:
        return True
    else:
        return False

print(isValid("({{{{}}}))"))